src/main/kotlin/org/rust/ide/actions/RustFmtFileAction.kt

```diff
@@ -0,0 +1,60 @@
+package org.rust.ide.actions
+
+import com.intellij.execution.ExecutionException
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.module.ModuleUtilCore
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.PathUtil
+import org.rust.cargo.toolchain.RustToolchain
+import org.rust.ide.notifications.showBalloon
+import org.rust.lang.RustFileType
+
+class RustFmtFileAction : DumbAwareAction() {
+    override fun update(e: AnActionEvent?) {
+        super.update(e);
+        e ?: return
+        val project = e.project
+        val file = e.getData(CommonDataKeys.VIRTUAL_FILE)
+        val editor = e.getData(CommonDataKeys.EDITOR)
+
+        if (project == null || file == null || !file.isInLocalFileSystem() || !isAvailableOnFile(file) || editor == null) {
+            e.presentation.isEnabled = false
+            return
+        }
+        e.presentation.isEnabled = true
+    }
+
+    protected fun isAvailableOnFile(file: VirtualFile): Boolean {
+        return file.fileType === RustFileType
+    }
+
+    override fun actionPerformed(e: AnActionEvent?) {
+        val project = e?.project ?: return
+        val file = e?.getRequiredData(CommonDataKeys.VIRTUAL_FILE) ?: return
+
+        file.canonicalPath ?: return
+
+        val document = FileDocumentManager.getInstance().getDocument(file)
+        if (document != null) {
+            FileDocumentManager.getInstance().saveDocument(document)
+        } else {
+            FileDocumentManager.getInstance().saveAllDocuments()
+        }
+
+        val module = ModuleUtilCore.findModuleForFile(file, project) ?: return
+        val moduleDirectory = PathUtil.getParentPath(module.moduleFilePath)
+
+        try {
+            RustToolchain.suggest()?.cargo(moduleDirectory)?.reformatFile(file.path) ?: return
```

Comment:
`suggest` just guesses probable toolchaing for use in the UI. To get the real toolchain configured for project, use `module.project.toolchain`.